lib/buffer.js

```diff
@@ -458,8 +458,11 @@ Buffer.concat = function concat(list, length) {
 
   if (length === undefined) {
     length = 0;
-    for (i = 0; i < list.length; i++)
-      length += list[i].length;
+    for (i = 0; i < list.length; i++) {
+      if (list[i].length) {
```

Comment:
Is this an unrelated change?

Reply:
`Buffer.concat` would throw `NaN` error(which I think is 
 wrong error) when the `list` is `INVALID_ARG_TYPE` if this didn't change.

For example,  when `list` is an Array of number, `list[i].length` is `undefined`, then `length += list[i].length;` cause `length` to be `NaN`, then 
https://github.com/nodejs/node/blob/3fbf55a87c5c74fef93b16721834f9e270796cef/lib/buffer.js#L467
 would catch the `NaN` error but not `ERR_INVALID_ARG_TYPE`

![screen shot 2019-02-17 at 11 11 01 pm](https://user-images.githubusercontent.com/23313266/52915107-07cc5800-330b-11e9-9c33-3c3a0c0a742c.png)

Maybe see this test:

https://github.com/nodejs/node/blob/3fbf55a87c5c74fef93b16721834f9e270796cef/test/parallel/test-buffer-concat.js#L57-L65


Some advices ?

Reply:
Oh, I see, so without this change too, test-buffer-concat fails. Makes sense. Thanks.